test/integration/httpd_role_test.lua

```diff
@@ -0,0 +1,76 @@
+local t = require('luatest')
+local treegen = require('luatest.treegen')
+local server = require('luatest.server')
+local fun = require('fun')
+local yaml = require('yaml')
+
+local helpers = require('test.helpers')
+local httpd_role = require('roles.httpd')
+local mock_role = require('test.mocks.mock_role')
+
+local g = t.group()
+
+g.after_each(function()
+    httpd_role.stop()
+end)
+
+g.test_httpd_role_usage = function()
+    helpers.skip_if_not_tarantool3()
+
+    local config = {
+        credentials = {
+            users = {
+                guest = {
+                    roles = {'super'},
+                },
+            },
+        },
+        iproto = {
+            listen = {{uri = 'unix/:./{{ instance_name }}.iproto'}},
+        },
+        groups = {
+            ['group-001'] = {
+                replicasets = {
+                    ['replicaset-001'] = {
+                        instances = {
+                            ['instance-001'] = {},
+                        },
+                    },
+                },
+            },
+        },
+        roles_cfg = {
+            httpd = {
+                default = {
+                    listen = 13000,
+                },
+                additional = {
+                    listen = 13001,
+                }
+            },
+            mock_role = {
+                {
+                    id = 1,
+                },
+                {
+                    id = 2,
+                    name = 'additional',
+                },
+            },
+        },
+    }
+    local dir = treegen.prepare_directory({}, {})
+
+    local config_file = treegen.write_file(dir, 'config.yaml',
+        yaml.encode(config))
+    local opts = {config_file = config_file, chdir = dir}
+    g.server = server:new(fun.chain(opts, {alias = 'instance-001'}):tomap())
+
+    g.server:start()
+
+    httpd_role.apply(config.roles_cfg.httpd)
```

Comment:
Adding them to instance config section `roles = ['roles.http', 'test.mocks.mock_role']` should be enough.